### PR TITLE
Return document symbols from the documentSymbolProvider

### PR DESCRIPTION
**Pull request by akshita31**
Fixes: #2414

### CODE REVIEW

File: src/features/documentSymbolProvider.ts [hunk 1]
```diff
@@ -83,3 +74,9 @@ function toSymbolKind(kind: string): vscode.SymbolKind {
 
     return kinds[kind];
 }
+
+function toVscodeRange(range: protocol.V2.Range):  vscode.Range{
```
Comment (rchande) [inline src/features/documentSymbolProvider.ts]: I think we might have an implementation of this somewhere else already

File: src/features/documentSymbolProvider.ts [hunk 2]
```diff
@@ -14,43 +14,34 @@ import SymbolRangeNames = protocol.V2.SymbolRangeNames;
...
-        element.DisplayName,
-        toSymbolKind(element.Kind),
-        parentElementName,
-        new vscode.Location(vscode.Uri.file(fileName), vscodeRange)
-    );
+function createSymbolForElement(element: Structure.CodeElement): vscode.DocumentSymbol {
+    const fullRange = element.Ranges[SymbolRangeNames.Full];
+    const nameRange = element.Ranges[SymbolRangeNames.Name];
+    
+    return new vscode.DocumentSymbol(element.DisplayName, JSON.stringify(element.Properties), toSymbolKind(element.Kind), toVscodeRange(fullRange), toVscodeRange(nameRange));
```
Comment (rchande) [inline src/features/documentSymbolProvider.ts]: Can you share an example of where this gets presented visually so we can see how the detail is shown? I don't think you want a json string.

File: src/features/documentSymbolProvider.ts (hunk 1 above)
Comment (akshita31) [reply src/features/documentSymbolProvider.ts]: Flavors of this one are present, not exactly this one. However I added this method to that file itself. Now there are three methods that are essentially doing the same thing. Do you think we should refactor a bit here ?

File: src/features/documentSymbolProvider.ts (hunk 2 above)
Comment (akshita31) [reply src/features/documentSymbolProvider.ts]: detail

The detail is shown in the outline view. And yes you are right we should not put a json string. Should we use the filename instead ?

**Comment by akshita31** [comment]
Ok, adding the filename doesn't make it look good.
filename

Any suggestions ?

File: src/features/documentSymbolProvider.ts [hunk 3]
```diff
@@ -25,32 +26,23 @@ export default class OmnisharpDocumentSymbolProvider extends AbstractSupport imp
...
-        element.DisplayName,
-        toSymbolKind(element.Kind),
-        parentElementName,
-        new vscode.Location(vscode.Uri.file(fileName), vscodeRange)
-    );
+function createSymbolForElement(element: Structure.CodeElement, filename: string): vscode.DocumentSymbol {
+    const fullRange = element.Ranges[SymbolRangeNames.Full];
+    const nameRange = element.Ranges[SymbolRangeNames.Name];
+    
+    return new vscode.DocumentSymbol(element.DisplayName, filename, toSymbolKind(element.Kind), toRange3(fullRange), toRange3(nameRange));
```
Comment (DustinCampbell) [inline src/features/documentSymbolProvider.ts]: Is this still hierarchical? Do we need to populate the [`children`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L2569-L2572) property on [`DocumentSymbol`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#DocumentSymbol)? Or, does that happen automatically passed on the ranges that are passed in?

**Comment by rchande** [comment]
@oscarg933 Thanks for taking the time to review this PR. It looks like you marked it as "approved" several times without leaving any comments. Was that intentional? Did you have feedback you wanted to leave?

File: src/features/documentSymbolProvider.ts [hunk 4]
```diff
@@ -11,10 +11,11 @@ import * as vscode from 'vscode';
 import Structure = protocol.V2.Structure;
 import SymbolKinds = protocol.V2.SymbolKinds;
 import SymbolRangeNames = protocol.V2.SymbolRangeNames;
+import { toRange3 } from '../omnisharp/typeConvertion';
```
Comment (rchande) [inline src/features/documentSymbolProvider.ts]: Can we rename "typeConvertion" -> typeConversion? I'm tired of seeing that typo

**Pull request merged**